poly/sharing_new.go

```diff
@@ -0,0 +1,161 @@
+import "github.com/dedis/crypto/abstract"
+
+// XXX: Un/Marshalling functions are still missing.
+
+// PriShare represents an individual private share v = p(i).
+type PriShare struct {
+	I int             // Index of the private share
+	V abstract.Scalar // Value of the private share
+}
+
+// PriPoly represents a secret sharing polynomial.
+type PriPoly struct {
+	g      abstract.Group    // Cryptographic group
+	coeffs []abstract.Scalar // Coefficients of the polynomial
+}
+
+// NewPriPoly creates a new secret sharing polynomial for the cryptographic
+// group g, the secret sharing threshold t, and the secret to be shared s.
+func NewPriPoly(g abstract.Group, t int, s abstract.Scalar) *PriPoly {
+	// XXX: old PriPoly.Pick()
+	return nil
+}
+
+// Threshold returns the secret sharing threshold.
+func (p *PriPoly) Threshold() int {
+	return len(p.coeffs)
+}
+
+// SharedSecret returns the shared secret p(0), i.e., the constant term of the polynomial.
+func (p *PriPoly) SharedSecret() abstract.Scalar {
```

Comment:
Is this function required ? Usually if you create the `PriPoly`, you're supposed to know the secret that you put in right ?
But if you think it is, then what do you think about adding field `secret` to the `PriPoly` struct in order to avoid doing Lagrange interpolations each time you want to recover the secret ?
It keeps the secret in memory, but avoid useless computations.